src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -755,8 +755,8 @@ public string IdentitySelectString
 			{
 				if (identitySelectString == null)
 					identitySelectString =
-						Factory.Dialect.GetIdentitySelectString(GetTableName(0), GetKeyColumns(0)[0],
-																										IdentifierType.SqlTypes(Factory)[0].DbType);
+						Factory.Dialect.GetIdentitySelectString(GetKeyColumns(0)[0], GetTableName(0),
+						                                        IdentifierType.SqlTypes(Factory)[0].DbType);
```

Comment:
Maybe some external dialects requiring the use of these parameters may be broken by this fix, if they were "compensating" for the inversion while being used for mappings where only entities are using identity columns. That is why this fix should not be put in 5.1.1 in my opinion.